utils/geoutils.py

```diff
@@ -190,7 +205,21 @@ def check_gdf_load(gdf):
                 gdf, GEOM_POSSIBLE_NAMES="geometry", KEEP_GEOM_COLUMNS="NO"
             )
         try:
-            return gpd.read_file(gdf)
+            layer_name = fiona.listlayers(gdf)
+            if len(layer_name) == 1:
+                return gpd.read_file(gdf)
+            else: 
+                # multi_layers = gpd.GeoDataFrame()
+                for count, layername in enumerate(layer_name):
+                    if count == 0: 
+                        multi_layers = gpd.read_file(gdf, layer=layername)
+                    else: 
+                        geopkg = gpd.read_file(gdf, layer=layername)
+                        multi_layers = pd.concat(
+                            [multi_layers, geopkg],
+                            ignore_index=True
+                        )    
+                return multi_layers
```

Comment:
In general, I understand why it makes sense to know if the label gpkg is a multi-layer or not. But do we have label gpkg in practice? I thought we always have a single-layer single-class gpkg for training? If this is not true, then I need to adjust my tiling script too.

Reply:
We mustly use single-layer single-class, but we want to also support multi-classes and Pierre told me that the GPKG will be multi layer.